scrapers_next/hi/people.py

```diff
@@ -1,89 +1,139 @@
+        capitol_addr = f"{contact_list[0]}, {state_addr_base}"
+
+        cap_phone, cap_fax = [x.split(":")[-1].strip() for x in contact_list[1:3]]
+
+        # Website does not allow scraping of member emails, so a manual check
+        #   of all member emails confirmed accuracy of below solution.
+        email_user = last_name.lower()
+        for string in self.social_remove_regexes:
+            email_user = string.sub("", email_user)
+        email_start = {"House": "rep", "Senate": "sen"}
+        email = email_start[chamber] + email_user + "@capitol.hawaii.gov"
+
+        soc_handles = {}
+        soc_links = contact_info.getnext().xpath("a")
+        for link in soc_links:
+            href = link.get("href").lower()
+            dom, han = self.soc_domain_and_handle_regex.search(href).groups()
+            for string in self.social_remove_regexes:
+                han = string.sub("", han)
+            soc_handles[dom] = han
+
+        # Website provides bad youtube handle for Sen Mike Gabbard
+        if f"{first_name} {last_name}" == "Mike Gabbard":
+            soc_handles["youtube"] = "senmikegabbard"
 
         p = ScrapePerson(
-            name=data["first_name"] + " " + data["last_name"],
+            name=f"{first_name} {last_name}",
             state="hi",
-            chamber=chamber,
-            district=data["district"],
-            given_name=data["first_name"],
-            family_name=data["last_name"],
-            party=party,
-            email=data["email"],
+            chamber="lower" if chamber[0] == "H" else "upper",
+            district=district,
+            given_name=first_name,
+            family_name=last_name,
+            party="Democratic" if party == "D" else "Republican",
+            image=member_photo,
+            email=email,
         )
-        p.capitol_office.address = address
-        p.capitol_office.voice = data["voice"]
-        p.capitol_office.fax = data["fax"]
-        p.add_source(data["url"])
-        p.add_link(data["url"])
+
+        if leader:
+            p.extras["title"] = title
+
+        p.capitol_office.address = capitol_addr
+        p.capitol_office.voice = cap_phone
+        p.capitol_office.fax = cap_fax
+
+        p.add_source(self.source.url)
+        p.add_link(member_page)
+
+        if soc_handles.get("facebook"):
+            p.ids.facebook = soc_handles["facebook"]
+        if soc_handles.get("instagram"):
+            p.ids.instagram = soc_handles["instagram"]
+        if soc_handles.get("twitter"):
+            p.ids.twitter = soc_handles["twitter"]
+        if soc_handles.get("youtube"):
+            p.ids.youtube = soc_handles["youtube"]
+
+        # TODO: Update "PersonIdBlock" object in OS Core to include any other
+        #  possible social media ids --> then convert below code to use newly
+        #  available p.ids.[platform] method
+        social_ids = {"facebook", "instagram", "twitter", "youtube"}
+        for social in soc_handles.keys():
```

Comment:
```
for handle, value in soc_handles.items():
    if not value:
        continue
    if handle == "facebook":
        p.ids.facebook = value
    elif handle == "instagram":
        p.ids.instagram = value
    elif handle == "twitter":
        p.ids.twitter = value
    elif handle == "youtube":
        p.ids.youtube = value
    if handle not in social_ids:
        if "extra_social_ids" not in  p.extras:
            p.extras["extra_social_ids"] = [social]
        else:
            p.extras["extra_social_ids"].append(social)
```

Could maybe replace all the social work?

Reply:
Definitely could, but think that last `if` should make sure to actually capture the handle/value pair

Reply:
Yes, with the adjustment of replacing `social` with `value` in your above code block, that will be great. I will push it up momentarily. Thanks!

Reply:
Thank you, great replacement. It also made me realize I was handling the last conditionals wrong for `extra_social_ids`.

I took your suggested code block but fixed my logic for storing the `extra_social_ids` [in this solution](https://github.com/openstates/openstates-scrapers/pull/4247/commits/8b86fd9f2992cdc9d3fd8d9e40ea6d8d139843b8#diff-673087561dd45ad141eaa7c980527f29708f9be9c8ae73d0ee0b4ddb762f21a8R120-R139).